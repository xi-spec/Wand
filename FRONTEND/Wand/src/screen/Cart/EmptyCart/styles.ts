import { StyleSheet, Dimensions } from 'react-native';
import { COLORS, SIZE, FONTS } from '../../../constants/theme';

const { height } = Dimensions.get('screen');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: height * 0.2,
    alignItems: 'center'
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 50
  },

  text: {
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.second,
    paddingVertical: 20,
    marginHorizontal: 20
  },

  buttonsContainer: {
    width: '90%',
    height: 80,
    position: 'absolute',
    bottom: height * 0.20,
    backgroundColor: 'white'
  },
  button: {
    width: '100%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 10,
    paddingHorizontal: 10,
    borderRadius: 5

  },
  buttonText: {
    color: 'white',
    fontFamily: FONTS.primary,
    fontSize: SIZE.primary
  },
  colorButton: {
    backgroundColor: COLORS.primary
  },
  whiteButton: {
    borderWidth: 1

  },
  blackText: {
    color: 'black'
  }
});

export default styles;
