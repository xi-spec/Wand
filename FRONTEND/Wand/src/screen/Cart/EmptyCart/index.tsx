import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import styles from './styles';
import { IMAGES } from '../../../constants/images';
import { CART_TEXT } from '../../../constants/screenTexts';

export default function EmptyCart ({ navigation }:{navigation:{reset:Function, navigate:Function}}) {
  return (
        <View style={styles.container}>
          <Image style={styles.image}source={{ uri: IMAGES.emptyCart }}></Image>
        <Text style={styles.text}>{CART_TEXT.EMPTY_CART_TITLE}</Text>
        <Text style={styles.text}>{CART_TEXT.EMPTY_CART_SUBTITLE}</Text>

        <View style={styles.buttonsContainer}>

        <TouchableOpacity
        testID='shopNow'
        style={{ ...styles.button, ...styles.colorButton }}
        onPress={() => navigation.reset({
          index: 0,
          routes: [{ name: 'Home' }]
        })}>
            <Text style={styles.buttonText}>SHOP NOW</Text>
          </TouchableOpacity>

          <TouchableOpacity
             testID='favourite'
          style={{ ...styles.button, ...styles.whiteButton }}
           onPress={() => navigation.navigate('Favourite')}>
            <Text style={{ ...styles.buttonText, ...styles.blackText }}>VIEW WISHLIST</Text>
          </TouchableOpacity>
        </View>

      </View>

  );
}
