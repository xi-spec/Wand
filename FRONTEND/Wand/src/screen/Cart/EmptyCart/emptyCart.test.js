import { render, fireEvent, act } from '@testing-library/react-native';
import React from 'react';
import EmptyCart from './index';

describe('Given a EmptyCart component', () => {
  let navigation;
  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      reset: jest.fn()
    };
  });

  describe('When is rendered ', () => {
    test('Then is matched to snapshop', () => {
      const tree = render(<EmptyCart/>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When SHOP NOW button is pressed', () => {
    test('Then actions.increaseQuantityCart is invoked', () => {
      const { getByTestId } = render(<EmptyCart navigation={navigation}/>);

      act(() => {
        fireEvent.press(getByTestId('shopNow'));
      });

      expect(navigation.reset).toHaveBeenCalled();
    });
  });

  describe('When  VIEW WISHLIST button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getByTestId } = render(<EmptyCart navigation={navigation}/>);

      act(() => {
        fireEvent.press(getByTestId('favourite'));
      });

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
