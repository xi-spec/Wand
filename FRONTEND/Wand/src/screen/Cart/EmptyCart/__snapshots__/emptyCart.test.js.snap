// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a EmptyCart component When is rendered  Then is matched to snapshop 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 266.8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/60506d7b3b22568450a0dc85/e2f00d68e8c7e38709c459cd55d470ad/finall3-e1604082541825-400x400.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    YOUR CART IS EMPTY
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 14,
        "marginHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    Check if there any products in your wishlist and snatch them up before they're gone
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 266.8,
        "height": 80,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E3192",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="shopNow"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        SHOP NOW
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="favourite"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        VIEW WISHLIST
      </Text>
    </View>
  </View>
</View>
`;
